### PR TITLE
[ZEE-9665] Added PowerQueryReference and related classes

### PR DESCRIPTION
**Pull request by paul-stanley**
JIRA issue: https://actian.atlassian.net/browse/ZEE-9665

##### PR summary:

- New classes to associate Power Queries with datasets, to replace passing the Power Query to the platform via a property.
- This will be utilized by Fabric/PowerBI connector to pass the Power Query + DSN details that come from the connector's configuration.

##### Checklist for Developer:

- [ x ] Summary part has been documented
- [ x ] Reviewers have been requested
- [ ] Reviews & comments have been taken into consideration
- [ ] Commits have been reworded and history cleaned

##### Checklist for Reviewer

- [ ] Code has been reviewed, commented and changes have been requested
- [ ] Reviews & comments have been taken into consideration

### CODE REVIEW

File: src/main/java/zeenea/connector/common/powerquery/OdbcEngine.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  /**
+   * Returns the OdbcEngine associated with the given name. Returns UNKNOWN if the name is null or
+   * not found.
+   *
+   * @param name The name to look up
+   * @return The matching OdbcEngine or UNKNOWN
+   */
+  public static OdbcEngine fromName(String name) {
+    if (name == null) {
```
Comment (sebastien-fauvel-zeenea) [inline src/main/java/zeenea/connector/common/powerquery/OdbcEngine.java]: Technically, it's useless because the code above return UNKOWN with null value,
but it's probably better to keep it like that to be more explicit on this edge case.

File: src/test/java/zeenea/connector/common/powerquery/PowerQueryReferenceTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    assertEquals(
+        "let source = Excel.Workbook(File.Contents(\"C:\\data.xlsx\")) in source",
+        powerQueryReference.getPowerQuery());
+    assertEquals(List.of(), powerQueryReference.getDsns());
+  }
+
+  @Test
+  @DisplayName("PowerQueryReference builder should handle valid inputs")
+  void shouldCreatePowerQueryReferenceWithBuilder() {
+    OdbcDsn odbcDsn =
```
Comment (sebastien-fauvel-zeenea) [inline src/test/java/zeenea/connector/common/powerquery/PowerQueryReferenceTest.java]: We never really use values inside the OdbcDsn object. 
It may simplify tests if we call a fonction to create one (with just name as parameter when we need two differents)

**Review by florenta-zeenea — APPROVED** [review]

**Review by sebastien-fauvel-zeenea — APPROVED** [review]
